src/Workspaces/CSharp/Portable/CodeGeneration/NamedTypeGenerator.cs

```diff
@@ -79,7 +79,7 @@ internal static class NamedTypeGenerator
 
             var members = GetMembers(namedType).Where(s => s.Kind != SymbolKind.Property || PropertyGenerator.CanBeGenerated((IPropertySymbol)s))
                                                .ToImmutableArray();
-            if (namedType.IsRecord)
+            if (namedType.IsRecord && namedType.IsReferenceType)
```

Comment:
i wouldn't make this change.  looking at the code it looks like it would handle struct records jsut fine.

Reply:
I made the change because of the cast below which would crash for record structs. Record structs use `RecordStructDeclarationSyntax` which derives from `TypeDeclarationSyntax` but isn't related to `RecordDeclarationSyntax`.

---
In reply to: [609211082](https://github.com/dotnet/roslyn/pull/52464#discussion_r609211082) [](ancestors = 609211082)